specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-04-27-preview/desktopvirtualization.json

```diff
@@ -0,0 +1,5375 @@
+            "Automatic",
+            "Direct"
+          ],
+          "x-ms-enum": {
+            "name": "PersonalDesktopAssignmentType",
+            "modelAsString": true
+          },
+          "type": "string",
+          "description": "PersonalDesktopAssignment type for HostPool."
+        },
+        "loadBalancerType": {
+          "enum": [
+            "BreadthFirst",
+            "DepthFirst",
+            "Persistent"
+          ],
+          "x-ms-enum": {
+            "name": "LoadBalancerType",
+            "modelAsString": true
+          },
+          "type": "string",
+          "description": "The type of the load balancer."
+        },
+        "ring": {
+          "type": "integer",
+          "description": "The ring number of HostPool."
+        },
+        "validationEnvironment": {
+          "type": "boolean",
+          "description": "Is validation environment."
+        },
+        "registrationInfo": {
+          "$ref": "#/definitions/RegistrationInfoPatch",
+          "description": "The registration info of HostPool."
+        },
+        "vmTemplate": {
+          "type": "string",
+          "description": "VM template for sessionhosts configuration within hostpool."
+        },
+        "ssoadfsAuthority": {
+          "type": "string",
+          "description": "URL to customer ADFS server for signing WVD SSO certificates."
+        },
+        "ssoClientId": {
+          "type": "string",
+          "description": "ClientId for the registered Relying Party used to issue WVD SSO certificates."
+        },
+        "ssoClientSecretKeyVaultPath": {
+          "type": "string",
+          "description": "Path to Azure KeyVault storing the secret used for communication to ADFS."
+        },
+        "ssoSecretType": {
+          "enum": [
+            "SharedKey",
+            "Certificate",
+            "SharedKeyInKeyVault",
+            "CertificateInKeyVault"
+          ],
+          "x-ms-enum": {
+            "name": "SSOSecretType",
+            "modelAsString": true
+          },
+          "type": "string",
+          "description": "The type of single sign on Secret Type."
+        },
+        "preferredAppGroupType": {
+          "enum": [
+            "None",
+            "Desktop",
+            "RailApplications"
+          ],
+          "x-ms-enum": {
+            "name": "PreferredAppGroupType",
+            "modelAsString": true
+          },
+          "type": "string",
+          "description": "The type of preferred application group type, default to Desktop Application Group"
+        },
+        "startVMOnConnect": {
+          "type": "boolean",
+          "description": "The flag to turn on/off StartVMOnConnect feature."
+        },
+        "sessionHostConfiguration": {
+          "$ref": "#/definitions/SessionHostConfigurationProperties",
+          "description": "The session host configurations of HostPool."
+        }
+      }
+    },
+    "SessionHostConfigurationProperties": {
```

Comment:
missing OS DiskType?

Reply:
also missing: PostUpdateTemplateURL

Reply:
> also missing: PostUpdateTemplateURL

we're not adding this anymore. if needed we'll add later